homeassistant/components/meraki/device_tracker.py

```diff
@@ -43,8 +48,22 @@ def __init__(self, config, async_see):
         self.validator = config[CONF_VALIDATOR]
         self.secret = config[CONF_SECRET]
 
+    def validate_secret(self, secret):
+        """Validate shared secret."""
+        if not secret:
```

Comment:
We should move this check to the caller of this method. We shouldn't pass an empty secret or `False` as secret to this method.